src/script/components/HistoryImport/index.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {WebAppEvents} from '@wireapp/webapp-events';
+import {amplify} from 'amplify';
+import {FC, useEffect, useState} from 'react';
+
+import Icon from 'Components/Icon';
+import LoadingBar from 'Components/LoadingBar';
+
+import {getLogger} from 'Util/Logger';
+import {t} from 'Util/LocalizerUtil';
+import {formatDuration} from 'Util/TimeUtil';
+import {loadFileBuffer} from 'Util/util';
+import {WebWorker} from 'Util/worker';
+
+import {Config} from '../../Config';
+import {BackupRepository} from '../../backup/BackupRepository';
+import {CancelError, DifferentAccountError, ImportError, IncompatibleBackupError} from '../../backup/Error';
+import {MotionDuration} from '../../motion/MotionDuration';
+import {ContentViewModel} from '../../view_model/ContentViewModel';
+import renderElement from 'Util/renderElement';
+
+export enum HistoryImportState {
+  DONE = 'HistoryImportState.STATE.DONE',
+  IMPORTING = 'HistoryImportState.STATE.IMPORTING',
+  PREPARING = 'HistoryImportState.STATE.PREPARING',
+}
+
+interface HistoryImportProps {
+  readonly backupRepository: BackupRepository;
+  file: File;
+  onClose?: () => void;
+}
+
+const HistoryImport: FC<HistoryImportProps> = ({backupRepository, file, onClose}) => {
+  const logger = getLogger('HistoryImportViewModel');
+
+  const [historyImportState, setHistoryImportState] = useState(HistoryImportState.PREPARING);
+  const [hasError, setHasError] = useState<Error | null>(null);
```

Comment:
Its not a boolean value so it shouldn't start with `"has"`, `[error, setError]` is a way to go IMO 😄